plugins/node/opentelemetry-instrumentation-mysql2/src/instrumentation.ts

```diff
@@ -81,14 +79,18 @@ export class MySQL2Instrumentation extends InstrumentationBase<MySQL2Instrumenta
           if (moduleExports === undefined) return;
           const ConnectionPrototype: mysqlTypes.Connection =
             moduleExports.Connection.prototype;
-          this._unwrap(ConnectionPrototype, 'query');
-          this._unwrap(ConnectionPrototype, 'execute');
+          if (isWrapped(ConnectionPrototype.query)) {
+            this._unwrap(ConnectionPrototype, 'query');
+          }
+          if (isWrapped(ConnectionPrototype.execute)) {
+            this._unwrap(ConnectionPrototype, 'execute');
+          }
```

Comment:
I'm curious, was this added because of your added "initialization" test below? Was that resulting in `logger('no original to unwrap to -- has ' + name + ' already been unwrapped?')` console errors?

Reply:
yes, correct